bokeh/server/tornado.py

```diff
@@ -188,14 +190,30 @@ def __init__(self, applications, prefix, hosts,
 
             all_patterns.extend(app_patterns)
 
+            # add a per-app static path if requested by any app handler
+            # (there should be at most one that requests a static path)
+            static_paths = set(h.static_path() for h in app.handlers)
```

Comment:
this logic converting N static paths to one seems a little bit like it should be the concern of Application, which is the thing that aggregates handlers, rather than the concern of this tornado class.